### ISSUE TITLE
Identity Provider Fixes

### ISSUE THREAD

**Issue by hifabienne** [zitadel/zitadel#5654]
Some of the identity providers do have some bugs, that should be fixed.

```[tasklist]
### Acceptance Criteria
- [x] Context of the configuration (instance/org) is not always clear and does switch at some point. Show where the user is editing, and have a look that it stays
- [ ] If allow register on the idp is enabled, the general allow registration from the login policy should be ignored
- [x] I am able to configure all providers and the settings are shown correctly on the detail
- [x] Removing Identity Providers is sometimes not possible. (Existing generic oidc provider --> Errors.Instance.IDPConfig.NotExisting ) https://github.com/zitadel/zitadel/pull/5685
- [x] Microsoft: Tenant ID is not shown on the detail page (Backend issue)
- [x] Micorsoft: Scopes are not taken into account when changing, they are not shown in the detail
- [x] Microsoft: On the organization config the scopes are not shown (Backend issue) will be fixed by https://github.com/zitadel/zitadel/pull/5665
- [x] LDAP: Required fields are not correctly, sometimes I can't create the provider because create button is disabled
- [x] LDAP: If only one server/userfilter/user object class is in the input, take it without clicking the +
- [x] Generic OIDC: auto-creation, auto update, allow creation, allow linking enable/disable not possible
- [x] Generic OAuth: auto creation, auto update, allow creation, allow linking enable/disable not possible
```

Can we show on the organization configuration that a user has to send the org scope to trigger the config? This is one of the problems I see so often.

**Comment by adlerhurst** [comment]
line hight isn't the same for each input field and header: 
<img width="432" alt="image" src="https://user-images.githubusercontent.com/27845747/231227230-c782c7a0-5bf1-42be-b49e-9f6c3ffd6b86.png">

**Comment by adlerhurst** [comment]
@livio-a  the api design of the azure tenants too complicated we have to redesign that.

the tenant field uses the following message
```proto
message AzureADTenant {
    oneof type {
        AzureADTenantType tenant_type = 1;
        string tenant_id = 2;
    }
}
```

I have absolutely no guidance when I have to use which field in the one of.

My proposal is to change the message like following:
```proto
message AzureADTenant {
   message AzureADTenantCommon {} //TODO: do we need the TenantID here?
   message AzureADTenantOrganisation {} //TODO: do we need the TenantID here?
   message AzureADTenantConsumer {} //TODO: do we need the TenantID here?
    oneof type {
        AzureADTenantCommon = 1;
        AzureADTenantOrganisation = 2;
        AzureADTenantConsumer = 3;
    }
}
```
This would solve the problem that we currently don't know which tenant type we are working with and the api would get much friendlier to work with

**Comment by adlerhurst** [comment]
oh wow this clarifies a lot thanks @livio-a. I think we should also add the link as a comment. maybe as comment of the whole message?

**Comment by adlerhurst** [comment]
@livio-a
@peintnermax and I just discussed if it would clarify if we add an additional enum field for the tenant ID so that it's even more clear that index 0 of the tenant type is tenant id:

```
enum AzureADTenantType {
    AZURE_AD_TENANT_TYPE_TENANT_ID = 0;
    AZURE_AD_TENANT_TYPE_COMMON = 1;
    AZURE_AD_TENANT_TYPE_ORGANISATIONS = 2;
    AZURE_AD_TENANT_TYPE_CONSUMERS = 3;
}
```

**Comment by mffap** [comment]
It seems like registration must be allowed to create a new user from an external IDP. Not sure if this only happens if prompted for additional (mandatory) user information or also when all userinfo is provided by the external IDP.

Also there seems to be an issue with the TenantID configuration on Organization level when using AzureAD. It only works to configure that on instance level. 

@hifabienne fyi, from our discovery call

**Comment by hifabienne** [comment]
@adlerhurst @peintnermax I saw that some things have already been fixed? Is this issue up to date?

**Comment by peintnermax** [comment]
I'll make some component changes today to improve ux 👍

**Comment by peintnermax** [comment]
A hint that the context was changed is implemented with #5935
